query_exp_test.go

```diff
@@ -0,0 +1,159 @@
+		int64Points     []TsInt64Point
+		timestampPoints []TsTimestampPoint
+	)
+
+	BeforeEach(func() {
+		alias = generateAlias(16)
+		columnsInfo = []TsColumnInfo{}
+		columnsInfo = append(columnsInfo, NewTsColumnInfo("blob_column", TsColumnBlob), NewTsColumnInfo("double_column", TsColumnDouble), NewTsColumnInfo("int64_column", TsColumnInt64), NewTsColumnInfo("timestamp_column", TsColumnTimestamp))
+	})
+	JustBeforeEach(func() {
+		timeseries = handle.Timeseries(alias)
+
+		timestamps = make([]time.Time, count)
+		blobPoints = make([]TsBlobPoint, count)
+		doublePoints = make([]TsDoublePoint, count)
+		int64Points = make([]TsInt64Point, count)
+		timestampPoints = make([]TsTimestampPoint, count)
+		for idx := int64(0); idx < count; idx++ {
+			timestamps[idx] = time.Unix((idx+1)*10, 0)
+			blobPoints[idx] = NewTsBlobPoint(timestamps[idx], []byte(fmt.Sprintf("content_%d", idx)))
+			doublePoints[idx] = NewTsDoublePoint(timestamps[idx], float64(idx))
+			int64Points[idx] = NewTsInt64Point(timestamps[idx], idx)
+			timestampPoints[idx] = NewTsTimestampPoint(timestamps[idx], timestamps[idx])
+		}
+
+		err := timeseries.Create(24*time.Hour, columnsInfo...)
+		Expect(err).ToNot(HaveOccurred())
+		blobColumn = timeseries.BlobColumn(columnsInfo[0].Name())
+		doubleColumn = timeseries.DoubleColumn(columnsInfo[1].Name())
+		int64Column = timeseries.Int64Column(columnsInfo[2].Name())
+		timestampColumn = timeseries.TimestampColumn(columnsInfo[3].Name())
+
+		doubleColumn.Insert(doublePoints...)
+		blobColumn.Insert(blobPoints...)
+		int64Column.Insert(int64Points...)
+		timestampColumn.Insert(timestampPoints...)
+	})
+	AfterEach(func() {
+		timeseries.Remove()
+	})
+	Context("QueryExp", func() {
+		It("should work", func() {
+			query := fmt.Sprintf("select * from %s in range(1970, +10d)", alias)
+			q := handle.QueryExp(query)
+			result, err := q.Execute()
+			defer handle.Release(unsafe.Pointer(result))
+			Expect(err).ToNot(HaveOccurred())
+			for _, table := range result.Tables() {
+				for rowIdx, row := range table.Rows() {
+					for idx, column := range table.Columns(row) {
+						columnIdx := idx - 1
+						// get values with universal getter
+						point := column.Get()
+						switch valueType := point.Type(); valueType {
+						case QueryResultBlob:
+							value := point.Value()
+							Expect(err).ToNot(HaveOccurred())
+							Expect(value).To(Equal(blobPoints[rowIdx].Content()))
+						case QueryResultDouble:
+							value := point.Value()
+							Expect(err).ToNot(HaveOccurred())
+							Expect(value).To(Equal(doublePoints[rowIdx].Content()))
+						case QueryResultInt64:
+							value := point.Value()
+							Expect(err).ToNot(HaveOccurred())
+							Expect(value).To(Equal(int64Points[rowIdx].Content()))
+						case QueryResultTimestamp:
+							value := point.Value()
+							Expect(err).ToNot(HaveOccurred())
+							Expect(value).To(Equal(timestampPoints[rowIdx].Content()))
+						}
+						// get values with specific getters
```

Comment:
I would remove these ifs by getting inner statements out of the `for idx, column ...` loop and just explicitly access e.g. `table.Columns(row)[0]` etc..